drivers/adc/max11205/iio_max11205.c

```diff
@@ -0,0 +1,261 @@
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdlib.h>
+#include "max11205.h"
+#include "iio_max11205.h"
+#include "errno.h"
+#include "no_os_util.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+#define MAX11205_BIT_SCALE	15
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/**
+ * @brief Handles the read request for raw attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure. In case of success, the size of the read data is returned.
+ */
+static int max11205_iio_read_raw(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel, intptr_t priv)
+{
+	int ret;
+	int32_t val;
+	int16_t data_raw;
+	bool new_data_avail = false;
+	struct max11205_iio_dev *iio_max11205;
+	uint32_t timeout = NO_OS_GENMASK(31,0);
```

Comment:
Is there a point in using such high of a value for this timeout? Wouldn't `no_os_mdelay()` fit better?

Reply:
The timeout is used as an exit condition of the while loop such that an infinite loop is avoided in case something happens with the device and it does not work anymore (no data ready interrupts are generated anymore). The timeout is a relatively large value to make sure platform with high CPU frequencies also allow for a minimum data conversion time (for max11205b can be up to 73ms). Taking for example a 120MHz processor, it will take about 36 seconds*number of instructions needed to execute the loop to reach the timeout. Of course, it can be lowered but the main reason for using the timeout was to avoid an infinite while loop.
no_os_mdelay would not fit in this scenario because the idea is to read the data as soon as new data is available, without any further delays.

Reply:
The timeout value was lowered. The timeout will now take 100msec * number of instructions needed to execute the loop instructions.